server/cli/start.go

```diff
@@ -153,15 +155,55 @@ func runStart(cmd *commander.Command, args []string) {
 
 	isBootstrap := cmd.Name() == "init"
 	err = s.Start(isBootstrap)
-	defer s.Stop()
 	if err != nil {
 		log.Errorf("Cockroach server exited with error: %v", err)
 		return
 	}
+	defer s.Stop()
 
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, os.Kill)
+	signal.Notify(c, syscall.SIGINT, syscall.SIGKILL, syscall.SIGTERM)
```

Comment:
I've reverted to using os.Interrupt and os.Kill and added the SIGTERM below in a separate statement with a TODO.